pyresample/test/test_spherical_utils.py

```diff
@@ -0,0 +1,361 @@
+
+    intuple = (3, (1, 2))
+    with patch('pyresample.spherical_utils.int_items_to_tuples') as mypatch:
+        mypatch.side_effect = fake_int_items_to_tuples
+        res = merge_tuples(intuple)
+
+    assert res == (3, 1, 2)
+
+    intuple = (0, (1, 2), (3, (4, 5)))
+    with patch('pyresample.spherical_utils.int_items_to_tuples') as mypatch:
+        mypatch.side_effect = fake_int_items_to_tuples
+        res = merge_tuples(intuple)
+
+    assert res == (0, 1, 2, 3, 4, 5)
+
+    intuple = (0, (1, 2), (3, (4, 5)), (6,))
+    with patch('pyresample.spherical_utils.int_items_to_tuples') as mypatch:
+        mypatch.side_effect = fake_int_items_to_tuples
+        res = merge_tuples(intuple)
+
+    assert res == (0, 1, 2, 3, 4, 5, 6)
+
+    intuple = (0, (1, 2), (3, (4, 5)), (6,), )
+    with patch('pyresample.spherical_utils.int_items_to_tuples') as mypatch:
+        mypatch.side_effect = fake_int_items_to_tuples
+        res = merge_tuples(intuple)
+
+    assert res == (0, 1, 2, 3, 4, 5, 6)
+
+    intuple = (0,)
+    with patch('pyresample.spherical_utils.int_items_to_tuples') as mypatch:
+        mypatch.side_effect = fake_int_items_to_tuples
+        res = merge_tuples(intuple)
+
+    assert res == (0,)
+
+
+def test_int_items_to_tuples():
+    """Test the function to turn integer scalars in a tuple into tuples."""
+
+    res = int_items_to_tuples(((1,), (2, 3)))
+    expected = ((1,), (2, 3))
+    assert res == expected
+
+    res = int_items_to_tuples((1, (2, 3)))
+    assert res == expected
+
+    res = int_items_to_tuples((1, 2))
+    assert res == ((1,), (2,))
+
+
+def test_find_union_pairs():
+    """Test finding a union pair."""
+
+    listed_sets = dict(enumerate((SET_A, )))
+    this = GetNonOverlapUnionsBaseClass(listed_sets)
+    retv = this._find_union_pair(listed_sets)
+
+    assert retv is None
+
+    listed_sets = dict(enumerate((SET_A, SET_B)))
+    this = GetNonOverlapUnionsBaseClass(listed_sets)
+    retv = this._find_union_pair(listed_sets)
+
+    assert retv is None
+
+    listed_sets = dict(enumerate((SET_C, SET_D)))
+    this = GetNonOverlapUnionsBaseClass(listed_sets)
+    retv = this._find_union_pair(listed_sets)
+
+    assert retv == ((0, 1), set(range(10, 20)))
+
+    listed_sets = dict(enumerate((SET_A, SET_B, SET_D)))
+    this = GetNonOverlapUnionsBaseClass(listed_sets)
+    retv = this._find_union_pair(listed_sets)
+
+    assert retv == ((1, 2), {2, 4, 6, 8, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19})
+
+    listed_sets = dict(enumerate((SET_A, SET_B, SET_C, SET_D)))
+    this = GetNonOverlapUnionsBaseClass(listed_sets)
+    retv = this._find_union_pair(listed_sets)
+
+    assert retv == ((1, 3), {2, 4, 6, 8, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19})
+
+    listed_sets = {0: {1, 3, 5, 7, 9},
+                   4: {0, 10},
+                   (2, (1, 3)): {2, 4, 6, 8, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19}}
+
+    this = GetNonOverlapUnionsBaseClass(listed_sets)
+    retv = this._find_union_pair(listed_sets)
```

Comment:
All these tests use a private method `_find_union_pair`, is this the actual intended usage of the class? Should this be public?

Reply:
The Base Class was created to make tests easier (using python sets, rather than having to mock polygons)

> I kind of understand the use case, but not really sure how it is supposed to be used. Maybe that would be more clear with some documentation. I put a couple concerns as comments inline below.

Thanks. Good point! Thought of that too. Not so easy for you to follow. I will try add some documentation, or at least show use case code here in the PR.

Reply:
Thx for looking into it @djhoese I see I did put a small use case above. But I will think of adding some documentation!